test/unit/math/mix/mat/fun/append_array_test.cpp

```diff
@@ -1,396 +1,392 @@
 #include <stan/math/mix/mat.hpp>
-#include <stan/math/prim/mat/fun/Eigen.hpp>
 #include <gtest/gtest.h>
 
+using stan::math::var;
```

Comment:
This file is pretty involved. The entry points are:

```
TEST(MathFunctions, append_array_prim)
TEST(MathFunctions, append_array_rev)
TEST(MathFunctions, append_array_fwd)
TEST(MathFunctions, append_array_mix)
```

Basically they all call ```check``` with various combinations of argument types and ```check``` calls:

```checkv``` which checks appending std::vectors of those types works as expected
and
```checkvv``` which checks appending std::vectors of std::vectors of those types works as expected

There are no tests of the autodiff here. We're just testing that all the casting between various containers of different autodiff types work.

The lingo for the typedefs is:

v -> var
f -> fvar
d -> double

So ```fd``` is ```fvar<double>``` and ```ffv``` is ```fvar<fvar<double> >```

V is VectorXd, R is RowVectorXd, M is MatrixXd.

Vfd is a Vector of ```fvar<double>```.

Mffv is a Matrix of ```fvar<fvar<var> >```